scripts/listener.ts

```diff
@@ -119,68 +175,192 @@ async function getTokenLists() {
   data = data.filter((o) => o); //remove undefined
   return data;
 }
-console.log(`Connecting to ${network} on url ${url}...`);
-if (chainSupportedBy(network, SubstrateEvents.Types.EventChains)) {
-  SubstrateEvents.createApi(url, spec).then(async (api) => {
+
+// returns either the RabbitMQ config specified by the filepath or the default config
+function getRabbitMQConfig(filepath: string) {
+  if (typeof filepath == 'string' && filepath.length == 0) return config;
+  else {
+    try {
+      let raw = fs.readFileSync(filepath);
+      return JSON.parse(raw.toString());
+    } catch (error) {
+      console.error(`Failed to load the configuration file at: ${filepath}`);
+      console.error(error);
+      console.warn('Using default RabbitMQ configuration');
+      return config;
+    }
+  }
+}
+
+// TODO: implement check to make sure returned data is really a spec
+export async function getSubstrateSpecs(chain: EventSupportingChainT) {
+  let url: string =
+    process.env.NODE_ENV == 'production'
+      ? `https://commonwealth.im/api/getSubstrateSpec?chain=${chain}`
+      : `http://localhost:8080/api/getSubstrateSpec?chain=${chain}`;
+  console.log(`Getting ${chain} spec at url ${url}`);
+
+  let data: any = await fetch(url)
+    .then((res) => res.json())
+    .then((json) => json.result)
+    .catch((err) => console.error(err));
+
+  return data;
+}
+
+/**
+ * Sets up a listener
+ * @param network The name of the chain/network to listen for events on
+ * @param listenerArg An object containing the processed listener options
+ * @return subscriber A subscriber instance to the chosen network
+ */
+export async function setupListener(
+  network: EventSupportingChainT,
+  listenerArg: IListenerOptions
+): Promise<IEventSubscriber<any, any>> {
+  // start rabbitmq
+  let handlers;
+  if (argv.rabbitMQ) {
+    handlers = [new StandaloneEventHandler(), producer];
+  } else {
+    handlers = [new StandaloneEventHandler()];
+  }
+
+  console.log(`Connecting to ${network} on url ${listenerArg.url}...`);
+
+  if (chainSupportedBy(network, SubstrateEvents.Types.EventChains)) {
+    // TODO: this is hardcoded but ideally should be made explicit in a config file eventually
```

Comment:
This is also the sort of thing we could store in the database, if that makes sense.